baremetal/metal3machine_manager.go

```diff
@@ -1008,8 +1008,10 @@ func (m *MachineManager) nodeReuseLabelExists(_ context.Context, host *bmov1alph
 
 // getBMCSecret will return the BMCSecret associated with BMH.
 func (m *MachineManager) getBMCSecret(ctx context.Context, host *bmov1alpha1.BareMetalHost) (*corev1.Secret, error) {
-	if host == nil || host.Spec.BMC.CredentialsName == "" {
-		return nil, nil
+	if host == nil {
+		return nil, errors.New("Host is empty")
+	} else if host.Spec.BMC.CredentialsName == "" {
+		return nil, errors.New("Credentials secret is empty")
```

Comment:
Looks like this is still causing a unit test to fail. We will need to either change that unit test or revert and put nolint here as well

Reply:
I added nolint for now.